### ISSUE TITLE
v2.7

### ISSUE THREAD

**Issue by evil-morfar** [evil-morfar/RCLootCouncil2#31]
Just a thread to discuss some changes for 2.7

I changed the layout/text for the descriptions in cdb4600fac0f60f61ef2c9c2f78c8b7a169d8be2, but I'm not sure if it could be better still. Any inputs @SafeteeWoW ?

**Comment by SafeteeWoW** [comment]
Look good for me.

**Comment by SafeteeWoW** [comment]
Found a bug:

Because we set the equipLoc of the armor token, ```:AutoPassCheck``` no longer works for the cloak armor token because of the first line of this function.

```
function RCLootCouncil:AutoPassCheck(subType, equipLoc, link, isToken, isRelic)
	if not tContains(autopassOverride, equipLoc) then
```
This breaks backward compatibility.
We have 2 solutions:
1. Break compatibility and fix the ```:AutoPassCheck```
2. Dont break it and ML does not set the ```equipLoc``` of armor type and we change ```:GetItemTypeText()```.

**Comment by evil-morfar** [comment]
I haven't really decided if we should break backward compability this version or wait. The list of "nice-to-have-breaking-changes" is getting bigger though - I just want to make sure to cover as many as possible so I won't have to do again next update.

As for the "don't break con": we don't save nor use it in the history anyway, and we could with relative ease restore the equipLoc client side in say ```:LocalizeLootTable()```.

**Comment by evil-morfar** [comment]
As for the release schedule of v2.7: I hope to have an alpha version up by tomorrow. Blizzcon is next week, and everyone thinks patch 7.3.5 drops by the reset after that, i.e. 7-8 november; that would make an opportune moment for 2.7 to be released. But I'd probably prefer to release next week to have some time for potential errors before 7.3.5.

**Comment by SafeteeWoW** [comment]
Change the frame height of loot frame to 80 is not good for some locale. It looks good for English.
No vertical space in Chinese Simplified (zhCn)
![wowscrnshot_102517_111700](https://user-images.githubusercontent.com/30888549/32015517-2d0f1c48-b976-11e7-9fc4-b60f37bba213.jpg)

**Comment by evil-morfar** [comment]
Back to 85 it is.

**Comment by evil-morfar** [comment]
How's b58bbdddcbeb0380125e64f02770780cb2ae8072 looking?

Here's the english:
![image](https://user-images.githubusercontent.com/10189576/32019179-a463859e-b9cc-11e7-93d9-363c20e9ce63.png)

Still using 80px entry height, but with the icon scaled to 78% it gives more space: `80*0,78 = 62,4` vs `85*0,73 = 62,02`.

**Comment by SafeteeWoW** [comment]
Looks fine for zhCN. testing other locale. Can you test it on other locale too? Not hard to switch and install other language in Battle.net.

**Comment by SafeteeWoW** [comment]
Don't forget to fix ```RCLootCouncil:AutoPassCheck```, as mentioned above.

**Comment by SafeteeWoW** [comment]
#34 #35

**Comment by evil-morfar** [comment]
I could, but afaik it's only zhCN/zhTW that special due to it's different font sizes.

**Comment by SafeteeWoW** [comment]
Remove L["g1"] and L["g2"] is not a good idea. Just reverse that commit edae461

**Comment by SafeteeWoW** [comment]
Similar to loot history, can you add messages in 
```RCLootCouncilML:AddItem``` and ``` RCLootCouncilML:BuildMLdb```? I am planning to add entries to those two functions in the next version of RC-EPGP.

**Comment by evil-morfar** [comment]
Given the limited time frame till 7.3.5 I don't want to break backwards compability in v2.7. It really doesn't cause any problems nor any massive overhead all the changes considered.

**Comment by SafeteeWoW** [comment]
Just let you be aware that sometimes the filter texts is colored, even if nothing is filtered. Happened before you merged my PRs today.

**Comment by SafeteeWoW** [comment]
Saw some long standing feature requests on the curseforge. Will you implement the feature to enable RC with ML? I think it's not hard to do. The addon just needs to treat the group leader as the ML if the loot method is not master loot and several option need to be added.(And this does not break backward compatibility, because guild using personal loot don't install old versions of RCLootCouncil.

**Comment by SafeteeWoW** [comment]
In RCLootCouncil-EPGP, I plan to add text "GP:xxx" in the item level text of the Loot Frame? How should I implement it?

**Comment by evil-morfar** [comment]
### Filters

I haven't been able to replicate that. Can you consistently replicate it?

### Override

Take a look at 8930f24c6a6110769c98bf304612cf13c89a2a04. I did prepare for it, but never got around to doing much with it, as it seemed weird, with a lot of potential for breaking stuff. As far as i remember, this did work as intended though.

### LootFrame ilvltext
If the item link is all you need to extract the GP value, then it's available in the arguments of `Entry:Update(entry, item)` with `item.link`. Anywhere else you can get it with `entry.item.link`. In fact, all the item info delivered to lootFrame is available in both.

I think the easiest way though, is to extract the entry from `EntryManager:GetEntry`, and then just edit `entry.itemLvl` directly. This needs to be a posthook though, as it would otherwise be overridden by the original call.

**Comment by SafeteeWoW** [comment]
### Override
If you add a variable ```override```. There is a lot of potential to break stuff because it adds complexity. I mean that ```addon.IsMasterLooter= if i am the group leader; addon.masterLooter=group leader``` when the loot method is not master loot.
I think you just to modify ```:NewMLCheck, :GetML, :OnRaidEnter``` a bit, and add several option.

**Comment by SafeteeWoW** [comment]
Can you change the note icon on the lootframe a bit, so it looks different after note is done?

**Comment by evil-morfar** [comment]
#### Filters
So it happens often? I don't really see how this can happen, nor have I seen it myself yet. 

#### Override
Doing what you're suggesting would always make the group leader the ML, which affects stuff like the popup when entering a dungeon, and depending on settings, auto enable RCLootCouncil on every dungeon. I'm pretty sure there's some other things as well, but I can't remember any on top of my head  - I do remember, though, that an opt in was the "best" way to do it when I started to implement it.

#### ilvl text
Well, after I changed the way the lootTable is handed to votingFrame, the lootTable in core.lua and votingframe.lua is the same table, so `:GetLootTable()` should just be moved to core.lua, just to make it more accesible and not being dublicated.

#### Note
Yeah - any ideas? The only one I can think of right now is to desaturate it.

#### 2.7
How extensive have you tested #39, #43 and #44 ? I haven't had much time lately, so I'm thinking about doing a quick alpha release today or tomorrow, and then doing a release sunday, just to have a few days with a broader userbase for potential bugs before Antorus opens. Those 3 features are the only ones I'm nervous about adding without having raid tested them myself. Otherwise the might have to wait for the next update (this one is kinda huge already).

**Comment by SafeteeWoW** [comment]
All pull requests are tested extensively. I do need to test again after all are merged though.

**Comment by evil-morfar** [comment]
It does if you mouse over it. I agree there should be an indicator for when the noted is there, but I don't think it should be displayed all the time.

**Comment by SafeteeWoW** [comment]
It is inconvenient to not shown note at all times. When the content of note is important, I have to mouseover the note button is double-check it.

**Comment by SafeteeWoW** [comment]
Override

> Doing what you're suggesting would always make the group leader the ML, which affects stuff like the popup when entering a dungeon, and depending on settings, auto enable RCLootCouncil on every dungeon. I'm pretty sure there's some other things as well, but I can't remember any on top of my head - I do remember, though, that an opt in was the "best" way to do it when I started to implement it.

Just need to add several options to let the group leader to choose whether show popup when entering the dugeon/auto enable RC, when loot method is not master loot.

**Comment by SafeteeWoW** [comment]
I have implemented all remaining feature requests on RCLootCouncil on Curseforge. Hope we can see these features on v2.7

**Comment by SafeteeWoW** [comment]
I have to sleep now. If there is any issue, I will resolve those when I get up.

**Comment by evil-morfar** [comment]
I've decided to make the cut here - mainly based on my comments in individual PRs, but also due to the massive amount of changes there's already in the update. It's not good to add too much at once - it becomes overwhelming pretty fast. 

I'm planning to run the current test version until sunday and then release it. That'll give a few days with more testers before Antorus. I'm not home until sunday afternoon (GMT+1 btw), but feel free to leave your comments.

Also, a huge thanks for all your great work!

**Comment by SafeteeWoW** [comment]
+ Small bug fixes that should be in v2.7: #63, #61, #56 
+ Code cleanup for v2.7: #60, #58
+ New feature that should be in v2.7: #51
+ Not sure: #54, #57 
+ v2.8: Other

**Comment by SafeteeWoW** [comment]
LUA export does not have correct syntax.

**Comment by evil-morfar** [comment]
I haven't had much time lately, and after finding out Antorus isn't released until 28/11 I haven't been in a rush to finish 2.7. Hopefully I'll get some more time for it the coming days, and I still plan to release it at least a week before.

I still believe we should push some of the remaining "major" changes to 2.8 to avoid making an overwhelming update, and I pretty much agree with your list.

Feel free to suggest changes to the export, but I frankly don't see much room for improvement.

**Comment by SafeteeWoW** [comment]
#70, #71 are ready for v2.7

**Comment by SafeteeWoW** [comment]
I think #52 (Allow addon to work without ml) should be in v2.7 and that PR is really short and should be very easy to review.

**Comment by evil-morfar** [comment]
It's not really about the size nor review - my main concern is burying users with new features and changes - followed by how a change/addition might affect other parts of the addon (both now and in the future). I trust you've tested your work, so I'm not that concerned about the latter, although I still like to review everything.

I have been considering just shipping everything you've made, since it's ready, and to pack all major ui changes in one update.

**Comment by SafeteeWoW** [comment]
Encountered a rare error. I believe I have reported this to you before, got this again randomly just now.
```
2x ...s\RCLootCouncil\Libs\LibWindow-1.1\LibWindow-1.1-8.lua:128: attempt to perform arithmetic on a nil value
...s\RCLootCouncil\Libs\LibWindow-1.1\LibWindow-1.1-8.lua:128: in function `SavePosition'
RCLootCouncil\core.lua:2009: in function <RCLootCouncil\core.lua:2006>
```
The line reports bug is the following:
**if frame.minimized then frame:Maximize() else frame:Minimize() end**

**Comment by evil-morfar** [comment]
I think 29b346de68c87fbffd52bd06a738e0068d4fdfc9 will fix it - I forgot to add that after your last error.

I'd still like your feedback on whether to add everything to 2.7 or not.

**Comment by SafeteeWoW** [comment]
I agree to put everything in v2.7, but we would need more beta version before release it, at least I need to test the release candidate version in the real raid

**Comment by SafeteeWoW** [comment]
```
-- Check if council is received
if not tContains(self.council, self.masterLooter) then
    self:Debug("Received loot table without ML in the council", sender)
```
I know this is a cleaner code to check this, but I don't like to do raw comparison to the name, especially when wrong capitalization locks the user in an infinite loop.

**Comment by SafeteeWoW** [comment]
I don't think disabling the rightclick menu for awarded items is a good idea. There are some buttons in the rightclick menu that should be usable when the items have been awarded.

**Comment by SafeteeWoW** [comment]
Just note that I found this API
~~~``` IsArtifactRelicItem(itemID)```. This can simplify code ~~~
Nah, that is a bad API which requires caching.

**Comment by SafeteeWoW** [comment]
23570c4
I think note on the response button enables double check before sending the note, which is good feature to good.

**Comment by SafeteeWoW** [comment]
I also want to add an award later button in the right click menu

**Comment by evil-morfar** [comment]
#### v2.7
While I don't like doing such a big release, I think it's better than having two big (ish) releases within two weeks of each other. I'm starting to merge the remaining PRs now. Adding support for reawarding an already awarded item should probably be delayed for another release, as I imagine it requiring some structural changes. Also do note I was only talking about awarding again (not the same) in my previous comments, just in case that wasn't clear.

#### Notes on buttons
I don't agree - in fact I found it very annoying having it sticking around the buttons.

#### Award later
Could definitely be done - it would be as simple as adding a new entry to the menu and award it will nil winner. I'm just not sure I see a real need for it? Wouldn't people interested in awarding items later just do it instead of starting a session?

**Comment by SafeteeWoW** [comment]
Allow to change award in voting frame is very convenient though, and I have almost done it.

I found a bug and I need to make a decision.
```RCLootCouncilML:AddUserItem(item)``` does not add the item info ```self.lootInBags```, so I have to differ the bagged items that are added by ```RCLootCouncilML:AddUserItem(item)``` and award later.
My current idea is that :AddUserItem uses self:AddItem(item, true), ```bagged``` is boolean, while :SessionFromBags uses self:AddItem(item, i), ```bagged``` is a number to differ them

**Comment by SafeteeWoW** [comment]
I found a bug and I need to make a decision.
```RCLootCouncilML:AddUserItem(item)``` does not add the item info ```self.lootInBags```, so I have to differ the bagged items that are added by ```RCLootCouncilML:AddUserItem(item)``` and award later.
My current idea is that :AddUserItem uses self:AddItem(item, true), ```bagged``` is boolean, while :SessionFromBags uses self:AddItem(item, i), ```bagged``` is a number to differ them

---
And I want to add two simple commands:
/rc addAward: Add the item into ML.lootInBags for award later
/rc removeAward: Remove one item from ML.lootInBags for award later.

**Comment by SafeteeWoW** [comment]
I am working on PR #78 today. I didn't do much on other PRs

**Comment by evil-morfar** [comment]
I ment it would require alot of changes to the already changed structure - but feel free to do it.

There's no reason to differentiate them?
`lootInBags` are unawarded items that should be used to start a session at some point.
`awardedInBags` are items that have been through a session and awarded, that just needs to be given out. 

If a user wants to remind himself about items, he could do a "/rc add [item]" and click "Award Later" to have the item added to `lootInBags` thus having rc remember it.
Deleting items from `lootInBags` should be done with the delete button in the session frame (when doing "/rc start") - there's no need for commands for that imo.

**Comment by SafeteeWoW** [comment]
We have to differ them for the following code
```
-- ml_core.lua, master branch
if self.lootTable[session].bagged then   -- indirect mode (the item is in a bag)
	-- Add to the list of awarded items in MLs bags, and remove it from lootInBags
	tinsert(self.awardedInBags, {link = self.lootTable[session].link, winner = winner})
	tremove(self.lootInBags, session)
	awarded = true
```

**Comment by evil-morfar** [comment]
No - it should be added to `awardedInBags` when awarded - that doesn't mean it needs to exist in `lootInBags`. The `tremove` won't even produce an error. `lootInBags` is only for items we've looted from a boss that should be awarded at a later point.

**Comment by SafeteeWoW** [comment]
See my edited comment.

**Comment by evil-morfar** [comment]
I see what you mean now. It's true awarding an item (say session 1) from `AddUserItem()` would remove any existing session 1 from the `lootInBags`. Should be relatively easy to fix though.

**Comment by SafeteeWoW** [comment]
Have you done merging #44? If yes, I'll merge v2.7 branch in #78

**Comment by evil-morfar** [comment]
Somewhat, I'm having some issues with it - it doesn't seem to work. Haven't found out why yet.

**Comment by SafeteeWoW** [comment]
ok. I'm testing the current v2.7 branch now.

**Comment by SafeteeWoW** [comment]
Fix. #85

**Comment by evil-morfar** [comment]
> /rc add: This command name is confusing.

Do you have anything better in mind? It's definitely better than "addItem" as short commands == good commands.

**Comment by SafeteeWoW** [comment]
I don't bother this problem now.

**Comment by SafeteeWoW** [comment]
Can you merge #71?
I need to resolve the conflicts in other PRs before you can merge.

**Comment by SafeteeWoW** [comment]
I will merge #78 and #73 into one PRs when I think #78 is ready.

**Comment by evil-morfar** [comment]
Merged #71.

**Comment by SafeteeWoW** [comment]
Btw, I agree with your comments in #52 , I just didn't work on it yet.

**Comment by evil-morfar** [comment]
I'm working on it now, so I'll just add them in

**Comment by SafeteeWoW** [comment]
I cant make #78, #73, #53 ready today. My plan is test them individually tomorrow and merge them into a big new PR.

**Comment by evil-morfar** [comment]
Alright - isn't #73 ready though?

**Comment by SafeteeWoW** [comment]
Right, I just need to fix merge conflict to make #73 ready.
Doing it now

**Comment by SafeteeWoW** [comment]
I'll try to make #53 ready tonight and finish #78 tomorrow.

**Comment by SafeteeWoW** [comment]
#73 is ready

**Comment by SafeteeWoW** [comment]
#53 won't be ready today, because db stuffs requires some structure changes. I need to store the time when the item is added to expire the db properly. #53 will be combined into #78, which will be done tomorrow.

**Comment by SafeteeWoW** [comment]
7addbd3
This is not a good solution. You'd better force all columns to sort by session number.

**Comment by SafeteeWoW** [comment]
#86

**Comment by evil-morfar** [comment]
I'd prefer not to merge the different PRs together and just solve the merge conflicts when they appear, unless you really need it. It just makes for a better commit log.

**Comment by SafeteeWoW** [comment]
ok. Those PRs will be separate. Working on #53, which should be done torrmow.

**Comment by SafeteeWoW** [comment]
See you tomorrow

**Comment by SafeteeWoW** [comment]
Still working on it, almost done but haven't done any tests yet.

I think lots of localization needs to be adjusted or added to better explain the new features.

**Comment by evil-morfar** [comment]
I agree. I'll take a look at it when going through the changes.

**Comment by SafeteeWoW** [comment]
```
L["chat_commands"] = [=[
- config    - Open the options interface
- council   - Open the council interface
- history   - Open the Loot History (alt. 'h' or 'his')
- version   - Open the Version Checker (alt. 'v' or 'ver')
- open      - Open the voting frame
- reset     - Resets the addon's frames' positions
- test (#)  - Emulate a loot session with # items, 1 if omitted
- whisper   - Displays help to whisper commands
- add [item]- Add an item to the session frame
- award     - Start a session with items looted to your inventory
- winners   - Display the winners of awarded items looted to your inventory
- sync      - Open the synchronizer view
]=]
```
This should really be splitted into multipline local entries

**Comment by SafeteeWoW** [comment]
Do you know any API that get the information of ```"You may trade this item with players that were also eligible to loot this item for the next %s."``` in the tooltip?

The only way I thought is to parse the tooltip, but I also have to parse "hour" and "minute" properly, which is not that easy.

**Comment by SafeteeWoW** [comment]
We'd better give a new name to "master looter" in the addon's locale.
What should be the name?

**Comment by SafeteeWoW** [comment]
I think we should have a tutorial mode which prints extra text.

**Comment by SafeteeWoW** [comment]
LibWindow error happens again today.
```
4x ...s\RCLootCouncil\Libs\LibWindow-1.1\LibWindow-1.1-8.lua:128: attempt to perform arithmetic on a nil value
...s\RCLootCouncil\Libs\LibWindow-1.1\LibWindow-1.1-8.lua:128: in function `SavePosition'
RCLootCouncil\core.lua:2100: in function <RCLootCouncil\core.lua:2097>
```

**Comment by evil-morfar** [comment]
I just haven't had the time to change the chat commands locales. I'd like to to release 2.7 at the next reset if all goes well with the beta versions. I'll put in the locale changes, but I probably won't use them unless they're translated at the next reset.

As for the tradeability of items - it seems weird if there isn't any API to check for this, but I haven't been able to find any.

We're definitely not going to call it "Group Admin". You have to remember the majority of the users actually use ML and a sudden name change would alienate them. The addon is ment to be used with ML enabled - all the other features are just niche, that's only added to support the occasional user that wants to use it without ML. We've already done more than enough for them, and no matter what we do, it'll never be as useful as when using ML. 
Imo we've already exceeded the point where the features far extends the needs of the average user.

I don't like the idea of lots of prints for a tutorial. I once considered using Blizzard's tutorial overlay method, but I deemed it to big an addition for too little gain. The basics should be self explanatory (which, imo, it is). I did consider writing a guide/tutorial for the more advanced use cases, but noone have ever really asked about anything that couldn't be explained in an Faq fasion. With the amount of changes in 2.7 it is probably more needed than ever.

**Comment by SafeteeWoW** [comment]
Then probably some mention of master looter settings in RCLootCouncil can also apply to the group leader when the loot method is not master is enough

**Comment by SafeteeWoW** [comment]
I hope these PRs can be merged if everything is ok. I did change some locale, but probably I didn't make it much better. If possible, we should have beta3 and beta4 before release.

**Comment by evil-morfar** [comment]
I expect people to read the description on Curse. Some things might lack a proper explaination though, for which a tutorial would probably be nice. If people doesn't know about a feature, they're probably not interested in anything but the basics. Just typing "/rc" will also introduce the user to those features. I don't think an in game tutorial is worth it.

I'll do a beta3 today. #88 is on hold for now - I still haven't decided how I like it.

**Comment by SafeteeWoW** [comment]
Well, I just feel have to do '/rc add' manually when the loot method is not master feels incomplete, so I create that feature. It's common that guild don't use master loot, or basically cant because the group is not guild group.

**Comment by evil-morfar** [comment]
I really don't care much for people intentionally using PL - that's their decision, and they most likely have their reasons to do it. Those forced to use PL is another story though. If Blizzard hadn't made the whole guild ML thing, then I probably never would consider adding support for anything but ML.

**Comment by SafeteeWoW** [comment]
The feature of this PR will be changed. Instead of group leader sees everyone's loot in the session frame, raid member can link the item to the raid chat use a command in chat, in order to add the item to the session frame of the ML.

TLDR, This PR will alter "/rc add" so it non-ML can also use it and send the command to ML, depend on the setting of ML.
How is this? I will implement tomorrow if looks good for you

**Comment by SafeteeWoW** [comment]
Seems curseforge cant parse Beta3 changelog

**Comment by SafeteeWoW** [comment]
Did a master loot test, didn't find major bug. I just didn't test award later.

**Comment by SafeteeWoW** [comment]
The current manual roll feature is not synchronized across council and is not convenient, I will change that.

**Comment by SafeteeWoW** [comment]
Rarely, the loot frame becomes empty:
![wowscrnshot_111817_114146](https://user-images.githubusercontent.com/30888549/32984154-b667c232-cc55-11e7-8077-b27f4834b123.jpg)
There is error before I add nil check, but no error after I add nil check for libwindow:SavePostion
Any possible reason why this happens?

**Comment by SafeteeWoW** [comment]
We'd better print sth on login and ask the user to read the changelog

**Comment by evil-morfar** [comment]
#### PL
I'd probably prefer just to have it automatic - that's easier than teaching everyone in the raid how to do it, but I suppose there's pro's and cons for both.

#### Curse
Curse have once again changed their design without ensuring stuff's actually working. I'm really starting to hate them. I'll make a post on their forums.

#### Empty loot frame
This didn't happen before 2.7, as I tested that specific bug extensively last time I changed the loot frame.

#### Print on login
Yeah, probably.

**Comment by SafeteeWoW** [comment]
#### PL
I can make it so raid members gets a popup when he gets a tradable item, but it could be very annoying.

**Comment by SafeteeWoW** [comment]
I still want to have fully automatically loot handling feature when the loot method is not master.

After merging #88 and #94, whose feature is very related to master loot, loot handling in non-master loot can be very easily added.

**Comment by SafeteeWoW** [comment]
The other reason is I am against to put auto PL loot handling in another module is that it is not worth the effort to put it into another module. A lot more code is needed to ensure combatibility between modules and it is harder to maintain the module.

**Comment by SafeteeWoW** [comment]
Can you do a module up-to-date check in "verTest"? I will drop my code about verTest if you do.

**Comment by SafeteeWoW** [comment]
There is a bug after the profile of the addon is changed:
Error in Master Looter-> Buttons and Responses
Similar error in Extra Utilities.
```
12x RCLootCouncil\Modules\options-Options.lua:1364: attempt to compare number with nil
RCLootCouncil\Modules\options-Options.lua:1364: in function `member'
...nfig-3.0\AceConfigDialog-3.0\AceConfigDialog-3.0-64.lua:249: in function <...nfig-3.0\AceConfigDialog-3.0\AceConfigDialog-3.0.lua:197>
```
Simpliest and the most secure way to fix it will be doing a reload automatically when profile is changed.

**Comment by SafeteeWoW** [comment]
I am finishing the new version of RCLootCouncil-EPGP.
With the Antorus coming in few days, I will merge all my recent PRs into one tomorrow, if you are still busy.

**Comment by evil-morfar** [comment]
I'm not sure about adding module version checking to the ```verTest``` command. For that to be really useful it needs to send even more data (i.e. table of modules instead of a simple string) than it already does, which is already too much for a simple version check. I might change that at some point.

Do not merge the PRs. Except for potential merge conflicts, merging them does not make anything easier. The plan was to release 2.7 wednesday, and I've only been waiting for you to get epgp updated, and changes according to my comments on some of the PRs. If all goes well, I'll release it today.

**Comment by evil-morfar** [comment]
I'll release 2.7 at around 8:00 GMT+1 tomorrow (around 12hours from now). I think it's fair you get one more say before release since you made most of the changes.

**Comment by SafeteeWoW** [comment]
EPGP needs some more time to finish, I really want to add remaining PRs to RC.
Making a PR to verTest now, it is not hard to do it with by parsing the string.

**Comment by SafeteeWoW** [comment]
ilvl needs to be center aligned in session frame. Does not look good otherwise

**Comment by evil-morfar** [comment]
Except it looks awful when tokens with a + in their ilvl is also there

**Comment by SafeteeWoW** [comment]
Then prefix with some spaces

**Comment by evil-morfar** [comment]
You could argue a left align needs a space prefix. A prefixed center align doesn't work. What happens when ilvls changes? Next raid a ilvl 1000 item drops.

**Comment by SafeteeWoW** [comment]
To my knowledge, it's impossible that the ilvl 1000 legendary trinket drops by master looter. Therefore, all ilvl should have 3 digits. Even in the next expansion, because there is a ilvl squash, ilvl remains below 1000. It's likely ilvl stays with 3 digits in the near future.

**Comment by evil-morfar** [comment]
Center align still doesn't work well.

Eitherway, it's either both ML and PL or a questline. Doesn't really matter, I was only trying to put a 4 digit ilvl in perspective (i.e. 970+).

**Comment by SafeteeWoW** [comment]
The other solution will be dropping "+". It looks better in session fame and loot frame if all ilvls are 3 digits.

**Comment by SafeteeWoW** [comment]
Side note: I spell the word trade-able as "tradable" (American English), while British English tends to spell as "tradeable". Shouldn't matter though

**Comment by SafeteeWoW** [comment]
Offer version check in the next version of GroupGear and EU plz.

**Comment by evil-morfar** [comment]
I'll give you until tomorrow then.

While I don't really care, I think tradeable is more universally approved. 

Removing the + is not a good solution. Why is it so bad to left align it? I don't think it looks that bad, and a space before it will make it look just fine. 

EU and GG doesn't really need a version check at this point. I've also decided you're right - doing a module version check in the core addon is probably preferable.

**Comment by SafeteeWoW** [comment]
Reannounce and roll improvement #92 is ready. Can you have a look?

**Comment by SafeteeWoW** [comment]
Yes, left align it. I had some misunderstandings.

**Comment by SafeteeWoW** [comment]
If you are going to release it soon, can you upload the localization to Curse? I gonna work on the Chinese translation.

**Comment by evil-morfar** [comment]
A space, as in one. More than that and it might become too big, which is the reason spaces doesn't work with center align

**Comment by evil-morfar** [comment]
I uploaded the locale yesterday with the current merges. I won't do the release today as I only had time this morning and won't be home for the rest of the day

**Comment by SafeteeWoW** [comment]
When are you going to release it? We'd better release it before US gets Antorus raid

**Comment by evil-morfar** [comment]
My isp decided to fuck up, so I've been without net all day. I have a raid in 20 mins, but hopefully I'll get it done today - definitely before Tuesday.

**Comment by SafeteeWoW** [comment]
I have finished the code of RCLootCouncil-EPGP and I am doing the complete testing right now.

**Comment by SafeteeWoW** [comment]
There is a bug that the text of moreInfo of voting frame is out of frame after resizing.

**Comment by SafeteeWoW** [comment]
Side note: In RC v3.0, you should fix the name of Russian characters. It's very easy to make mistake in RC-EPGP when some times I need to remember to convert RC name into EPGP name, but sometime I shouldn't. I probably should make a character with special name to test this out.

**Comment by evil-morfar** [comment]
I'm still not sure if #88 should be in 2.7. #92 seems good enough to go.

The namefixing doesn't necessarily have to go into v3.0, it just breaks backward compability to fix it, and requires a council reset. As far as I'm aware the issue lies in my forcing of titlecases - it only capitalizes ascii characters. Afaik epgp uses the name returned by ```UnitName``` and equivilant. 

My plan for v3.0 is to use GUID's to handle players, with name only being used on displays.

**Comment by evil-morfar** [comment]
Anyway, I'll push 2.7 tonight, I just need to do some final tests.

**Comment by SafeteeWoW** [comment]
RC-EPGP should be ready. Just the changelog and README and translation take me some time.

**Comment by evil-morfar** [comment]
I have no issues with the more info frame when resizing.

There's another kinda huge issue though. Reawarding an item doesn't change the response of the previous winner. I tried something quick (from line 180 in votingframe.lua):
```lua
local oldWinner
for k, v in ipairs(lootTable) do
	if addon:ItemIsItem(v.link, lootTable[s].link) then
		if v.awarded or oldWinner then -- This is a reaward - restore old response.
			oldWinner = v.awarded or oldWinner
			self:SetCandidateData(k, oldWinner, "response", self:GetCandidateData(k, oldWinner, "real_response"))
		end
		self:SetCandidateData(k, winner, "real_response", self:GetCandidateData(k, winner, "response"))
		self:SetCandidateData(k, winner, "response", "AWARDED")
	end
end
lootTable[s].awarded = winner
```
However this will not work if the items are not awarded in order, and it will remove awarded response from 1 winner when two equal items are awarded to two different persons. Can you think of anything other than checking all candidates for "real_response"?

**Comment by evil-morfar** [comment]
I would also like to change the name "Group" from the reannounce and request roll dropdown to "Everyone" - thoughts?

**Comment by SafeteeWoW** [comment]
> I would also like to change the name "Group" from the reannounce and request roll dropdown to "Everyone" - thoughts?

Sure

**Comment by SafeteeWoW** [comment]
Let's use another approach.
Instead of change the response, let's modify SetCellResponse/ResponseSort to display the the response as "awarded". We don't modify the response.

**Comment by evil-morfar** [comment]
Doing it in SetCellResponse would only allow us to show 1 candidate as the winner, unless doing serveral loops. Anyway, found a way to do it effortlessly.

**Comment by SafeteeWoW** [comment]
Several loops is not bad though. When the it is a single loop and loop size is normally less than 10, it should not be considered as a costly operation.

**Comment by evil-morfar** [comment]
Several loops, per candidate, per update - that's too much.

**Comment by SafeteeWoW** [comment]
Line 187 should be swapped with line 186.
```
						end
						self:SetCandidateData(k, winner, "real_response", self:GetCandidateData(k, winner, "response"))
```

**Comment by evil-morfar** [comment]
Hmm. No and no.

**Comment by SafeteeWoW** [comment]
```
-- Check modules. Parse the strings.
for _, str in pairs(moduleData) do
```
Need to check if moduleData is nil before this line

**Comment by evil-morfar** [comment]
Luckily Curse rejected the tag, otherwise it would have been pushed by now. Any other last minute finds?

**Comment by SafeteeWoW** [comment]
I am not testing it, because I'm writing the documentation. Just randomly found this bug.

**Comment by evil-morfar** [comment]
It just went through. [The total tally is 5.932 changes (!)](https://github.com/evil-morfar/RCLootCouncil2/compare/2.6.1-Release...master). Let's try not to do such a massive update next time. Either way, thanks for your work so far.
